### PR TITLE
Update node_exporter v1.6.0

### LINKED ISSUE #4000: Update github.com/prometheus/node_exporter to v1.6.0
An update for `github.com/prometheus/node_exporter` (version `v1.6.0`) is now available. Version `v1.5.0` is currently in use.

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
Updates `node_exporter` to [v1.6.0](https://github.com/prometheus/node_exporter/releases/tag/v1.6.0). Some notable changes:

- New `softirqs` collector.
- New `netisr` collector.
- New options for the `perf` collector.
- A few collectors got NetBSD support (`uname`, `meminfo`, `cpu`).

#### Which issue(s) this PR fixes

Fixes #4000 

#### Notes to the Reviewer
Open question: do we usually communicate on our side when collectors get deprecated for the next major release? The `ntp` and `supervisord` collectors were marked as deprecated (only code change for each was logging a warning when they are enabled).

#### PR Checklist

- [X] CHANGELOG updated
- [X] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -597,13 +595,18 @@ require (
...
 	gopkg.in/warnings.v0 v0.1.2 // indirect
-	howett.net/plist v0.0.0-20181124034731-591f970eefbb // indirect
+	howett.net/plist v1.0.0 // indirect
 	inet.af/netaddr v0.0.0-20220811202034-502d2d690317 // indirect
 	k8s.io/kube-openapi v0.0.0-20230303024457-afdc3dddf62d // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
+require (
```
Comment (marctc) [inline go.mod:605]: Is there any reason that this has its own `require` section?

**Comment by marctc** [comment]
> Open question: do we usually communicate on our side when collectors get deprecated for the next major release? The `ntp` and `supervisord` collectors were marked as deprecated (only code change for each was logging a warning when they are enabled).

I'd say that if the next release of the node exporter will make breaking changes in the config (like removing fields), we could announce it in https://grafana.com/docs/agent/latest/static/upgrade-guide/

**Review by clayton-cornell — COMMENTED** [review]
Docs look OK

Which version of Agent is this applicable to?

File: go.mod (hunk 1 above)
Comment (spartan0x117) [reply go.mod:605]: Not sure how it was introduced (I only used `go get`), and `go mod tidy` doesn't clean those up: https://github.com/golang/go/issues/56471

I'll manually join them.

**Comment by spartan0x117** [comment]
> Which version of Agent is this applicable to?

@clayton-cornell This would be for the next release, so `v0.35.0` of the Agent. Would the `Upgrade Guide` change that Marc suggested be done at the same time as that release (so in a different PR)?

**Comment by clayton-cornell** [comment]
> Would the `Upgrade Guide` change that Marc suggested be done at the same time as that release (so in a different PR)?

I would split them and reference back to this PR.

**Review by marctc — APPROVED** [review]

**Review by clayton-cornell — APPROVED** [review]

**Comment by marctc** [comment]
@spartan0x117 after fixing the conflicts i think we can merge it, thanks!

**Pull request merged**